src/persist-client/src/stats.rs

```diff
@@ -169,6 +173,9 @@ impl PartStats {
 /// TODO: Add more stats here as they become necessary.
 #[derive(Debug)]
 pub struct SnapshotStats {
+    /// Metrics for the persist backing shard, so the caller can report any
+    /// necessary counters.
```

Comment:
This is a bit sketchy, but all the ways of exposing this in adapter feel pretty sketchy. Hopefully it goes away when we can get rid of the decode-errors counter, one way or another.